### DISCUSSION TITLE
Where in the node.js source code does TCPWrap::Listen get called?

### DISCUSSION THREAD

**Discussion by dSalieri** [nodejs/help#4632] [Q&A]
I read the node.js source code to understand how the library works.

For example, this code:
```js
net.createServer((s) => console.log("connection", s)).listen(3000);
```
The `listen` method from js will logically call [`TCPWrap::Listen`](https://github.com/nodejs/node/blob/821ffab0f78972d6e63bafa598b0c6d92550072b/src/tcp_wrap.cc#L275C6-L275C21). But if we look at the parameters of `TCPWrap::Listen` we will see this `const FunctionCallbackInfo<Value>& args`. It turns out that the call from js does not call `TCPWrap::Listen`, what happens then?

Where does the call to `TCPWrap::Listen` occur and what causes this operation?

**Comment by avivkeller** [comment]
@nodejs/net
